drivers/I2C.cpp

```diff
@@ -135,6 +140,53 @@ void I2C::unlock()
     _mutex->unlock();
 }
 
+int I2C::recover(PinName sda, PinName scl)
+{
+    DigitalInOut pin_sda(sda, PIN_INPUT, PullNone, 1);
+    DigitalInOut pin_scl(scl, PIN_INPUT, PullNone, 1);
+
+    // Read and verify if recovery is required
+    if (pin_scl == 1) {
+        if (pin_sda == 1) {
+            // Return successfuly as SDA and SCL is high
+            return 0;
+        }
+    } else {
+        // Return as SCL is low and no access to become master.
+        return I2C_ERROR_BUS_BUSY;
+    }
+
+    // Send clock pulses, for device to recover 9
+    pin_scl.mode(PullNone);
+    pin_scl.output();
+    int count = 9;
+    while (count--) {
+        pin_scl.mode(PullNone);
+        pin_scl = 0;
+        wait_us(5);
+        pin_scl.mode(PullUp);
+        pin_scl = 1;
+        wait_us(5);
+    }
+
+    // Send Stop
+    pin_sda.output();
+    pin_sda = 0;
+    wait_us(5);
```

Comment:
How accurate this wait needs to be?
Depending on target this `wait_us(5)` could take between 6-20 us
For more details see https://github.com/ARMmbed/mbed-os/issues/7045

Reply:
@maciejbocianski  - Wait time as per the I2C specs should be 5us min and max is not specified, so 6-20us should be fine.

https://www.nxp.com/docs/en/data-sheet/PCF8584.pdf? - Section `12 I2C-BUS TIMING SPECIFICATIONS`

Reply:
Timings used here are 
1. https://github.com/ARMmbed/mbed-os/pull/8109/files/a30da7e61e0433bbbc50059315114511ee55f993#diff-eee492ff5f391a35b3c99975d9f059f2R165 Clocking SCL
```
tLOW SCL LOW time 4.7 − −µs
tHIGH SCL HIGH time 4.0 − −µs
```
2. https://github.com/ARMmbed/mbed-os/pull/8109/files/a30da7e61e0433bbbc50059315114511ee55f993#diff-eee492ff5f391a35b3c99975d9f059f2R175 - Stop condition setup

`tSU;STO STOP condition set-up time 4.0 − −µs`

Reply:
I think these timings should be just fine.
Longer is OK.

Also, this same sequence has been part of our RF driver code for a year now https://github.com/ARMmbed/atmel-rf-driver/pull/54 without any hickups. So I'm starting to believe that it is working.